From: Alessandro Carminati <acarmina@redhat.com>
Subject: kunit: fix compilation error on s390
Date: Sat, 29 Mar 2025 15:03:20 +0000

The current implementation of suppressing warning backtraces uses
__func__, which is a compile-time constant only for non -fPIC compilation.
GCC's support for this situation in position-independent code varies
across versions and architectures.

On the s390 architecture, -fPIC is required for compilation, and support
for this scenario is available in GCC 11 and later.

Link: https://lkml.kernel.org/r/20250329150320.331018-1-acarmina@redhat.com
Signed-off-by: Alessandro Carminati <acarmina@redhat.com>
Acked-by: David Gow <davidgow@google.com>
Cc: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/kunit/Kconfig |    1 +
 1 file changed, 1 insertion(+)

--- a/lib/kunit/Kconfig~bug-kunit-core-support-for-suppressing-warning-backtraces-fix
+++ a/lib/kunit/Kconfig
@@ -17,6 +17,7 @@ if KUNIT
 
 config KUNIT_SUPPRESS_BACKTRACE
 	bool "KUnit - Enable backtrace suppression"
+	depends on (!S390 && CC_IS_GCC) || (CC_IS_GCC && GCC_VERSION >= 110000)
 	default y
 	help
 	  Enable backtrace suppression for KUnit. If enabled, backtraces
_
